collector/library/pod.js

```diff
@@ -56,51 +46,44 @@ class Pod {
     }
 
     async checkAndProcessNewBatches() {
-        if (this.dead === true) {
-            return;
-        }
+        while (true) {
```

Comment:
this is not a bug but a little strange - you have to trust that if you are dead no one will call you again - to be more clear you could say `while(!this.dead) {` and remove the following If statement

Reply:
This is actually better than the current `while (true)`